File: 2022/0210_1447.最简分数.cpp
/*
 * @lc app=leetcode.cn id=1447 lang=cpp
 *
 * [1447] 最简分数
 */

// @lc code=start
class Solution {
public:
    vector<string> simplifiedFractions(int n) 
    {
        vector<string> res;
        for (int i = 2; i <= n; ++i)    // 分母i
        {
            for (int j = 1; j < i; ++j) // 分子j
            {
                if (__gcd(i, j) == 1)  
                // __gcd(x, y)函数:用于求x，y的最大公约数；x,y不能是浮点数
                {
                    res.push_back(to_string(j) + "/" + to_string(i));
                    // push_back:函数将一个新的元素加到vector的最后面，位置为当前最后一个元素的下一个元素
                    // to_string:将数字常量转换为字符串
                }
            }
        }
        return res;
    }
};
// @lc code=end

File: 2022/0210_1447.最简分数.py
#
# @lc app=leetcode.cn id=1447 lang=python3
#
# [1447] 最简分数
#

# @lc code=start
# 分母i的范围是[2, n], 分子j的范围是[1, i - 1]
# 先求分子分母的最大公约数，约分后通过字符串判重
# Python的math模块gcd函数运用：gcd(x, y)= == 1，最大公约数为1时，x/y为最简分数
class Solution:
    def simplifiedFractions(self, n: int) -> List[str]:
        ans = []
        for i in range(2, n + 1):
            for j in range(1, i):
                if gcd(i, j) == 1:
                    ans.append(str(j) + '/' + str(i))
        return ans
# @lc code=end

